docs/tutorials/stored-proc-tutorial.md

```diff
@@ -0,0 +1,89 @@
+
+
+:::warning
+Because stored procedures run inside the database, Only raw database queries are supported. 
+Query builders like [Knex.js](https://knexjs.org/) and ORMs like [TypeORM](./using-typeorm.md) and [Prisma](./using-prisma.md) are not supported in stored procedure functions.
+:::
+
+```javascript
+export interface dbos_hello {
+  name: string;
+  greet_count: number;
+}
+
+export class Hello {
+
+  @StoredProcedure()  // Run this function as a stored procedure
+  static async helloTransaction(ctxt: StoredProcedureContext, user: string) {
+    // Retrieve and increment the number of times this user has been greeted.
+    const query = "INSERT INTO dbos_hello (name, greet_count) VALUES (?, 1) ON CONFLICT (name) DO UPDATE SET greet_count = dbos_hello.greet_count + 1 RETURNING greet_count;"
+    const { rows } = await ctxt.query<dbos_hello>(query, [user]);
+    const greet_count = rows[0].greet_count;
+    return `Hello, ${user}! You have been greeted ${greet_count} times.\n`;
+  }
+}
+```
+
+## Deploying DBOS Stored Procedures
+
+DBOS Stored procedure functions depend on [PLV8](https://plv8.github.io/), a trusted JavaScript language extension for PostgresSQL. 
+PLV8 is supported out of the box on DBOS Cloud and several major cloud database providers.
+For running locally, we recommend using the [`sibedge/postgres-plv8` Docker image](https://plv8.com) provided by [Sibedge](https://sibedge.com/).
+For using DBOS Stored Procedures on your own PostgreSQL server, please see the [official PLV8 documentation](https://plv8.github.io/#building) for installation instructions.
+
+:::info
+As of DBOS 1.17, the [@dbos-inc/create](../api-reference/cli#npx-dbos-inccreate) templates have been updated to use `sibedge/postgres-plv8` in the `start_postgres_docker.js` script.
+Older DBOS applications using Docker will need to switch their PostgreSQL image from `postgres:16.1` to `sibedge/postgres-plv8` manually to support Stored Procedures.
+:::
+
+Before running your DBOS application that uses stored procedures, you need to deploy those stored procedures to the database.
+To deploy your stored procedure functions to the database, you need the [DBOS Compiler](../api-reference/dbos-compiler.md). 
+Add the DBOS Compiler package as a devDependency of your app via NPM:
+
+```
+npm install --save-dev @dbos-inc/dbos-compiler
+```
+
+Once the DBOS Compiler is installed, you can use it to deploy the stored procedures to the database server specified 
+in the [`dbos-config.yaml` file](../api-reference/configuration.md) with the following command:
+
+```
+npx dbosc deploy
+```
+
+To can deploy your app's stored procedures to the database via [Database Schema Management](../cloud-tutorials/database-management.md#database-schema-management). 
+To do so, simply add a call to `npx dbosc deploy` to the `database.migrate` section of your app's [`dbos-config.yaml` file](../api-reference/configuration.md).
```

Comment:
Mention that configuring migrate is essential for Cloud deployment to work.

Reply:
How much dbos cloud should I be getting into for a dbos transact tutorial?